### PR TITLE
Addressed issues in OpenZeppelin Audit Report

### PR DESCRIPTION
**Pull request by matt-lough**
- Tokens are now issued to buyers using the standard `transfer` function, but are still locked for transferring for anyone other than the crowdsale for 7 days after the crowdsale ends
- Updated vesting amounts
- Added `claimTokens` function to retrieve random tokens or eth that are sent to the contract
- Gas optimizations
- Updated tests
- Added tests

Vesting Contract changes are outlined below in the comments

### CODE REVIEW

**Comment by Janther** [comment]
The last 4 commits have everything related to the vesting.

A vesting contract that complies with the following rules given proper constructor params:
  - Advisors:
    - 25m POLY
    - vested until August 23rd, 2018
    - we specify numbers/addresses
  - Founders:
    - 150m POLY
    - 1/4th vested until 1 year cliff
    - with 1/48th vest every month thereafter
    - allocated to multiple addresses that we specify.
  - Reserve:
    - 450m POLY
    - vested until April 18th, 2018
    - with 10m POLY released every month
    - sent to a single address

I also updated the solidity compiler, truffle version, and zeppelin-solidity version.

File: contracts/PolyMathToken.sol [hunk 1]
```diff
@@ -11,40 +13,73 @@ contract PolyMathToken is PausableToken, BurnableToken {
...
+  function isCrowdsaleAddressSet() public constant returns (bool) {
+    return (address(crowdsale) != address(0));
+  }
+
+  modifier crowdsaleNotInitialized() {
+    require(!isCrowdsaleAddressSet());
+    _;
+  }
+
+  function PolyMathToken(address _presale_wallet) public {
```
Comment (Everhusk) [inline contracts/PolyMathToken.sol]: Why is the presale wallet specified as a single address? The requirement is to be able to upload a list of allocations for ethereum addresses to claim.

File: contracts/PolyMathVesting.sol [hunk 2]
```diff
@@ -1,45 +1,99 @@
...
+    cliffTime = _cliffTime;
     releaseTime = _releaseTime;
+    period = _period;
+    numPeriods = releaseTime.sub(startTime).div(period);
+  }
+
+  function allocateArray(
+      address[] _holders,
+      uint256[] _amounts
+  ) public onlyOwner {
```
Comment (Everhusk) [inline contracts/PolyMathVesting.sol]: This needs to be linked to the cap amounts to prevent allocations going over the caps specified

Comment (Janther) [reply contracts/PolyMathVesting.sol]: The allocations can't go higher than the tokens owned by the vesting contract.
This is checked in [line 66](https://github.com/PolymathNetwork/Crowdsale/pull/7/files#diff-4f658f7438e6d5ced4ff6ca77c6b9a0dR66)

**Pull request merged**